app/assets/javascripts/modules/nested-facets.js

```diff
@@ -0,0 +1,94 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  function NestedFacets (module) {
+    this.module = module
+    this.options = this.instantiateOptions()
+  }
+
+  NestedFacets.prototype.init = function () {
+    if (this.module) {
+      this.update()
+      this.module.addEventListener('change', () => {
+        this.update()
+      })
+    }
+  }
+
+  NestedFacets.prototype.update = function updateFacets () {
```

Comment:
I don't think you need to name these functions twice (unless you're doing something specific with that). So all of these prototype calls can be of the form

```JavaScript
NestedFacets.prototype.update = function () {
```